CMakeLists.txt

```diff
@@ -80,11 +72,18 @@ endif()
 add_subdirectory(svf)
 add_subdirectory(svf-llvm)
 
+include(GNUInstallDirs)
+install(
+  TARGETS SvfCore SvfLLVM
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
 install(
   DIRECTORY ${PROJECT_SOURCE_DIR}/svf/include/
             ${PROJECT_SOURCE_DIR}/svf-llvm/include/
   COMPONENT devel
-  DESTINATION include/svf
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/include/svf
   FILES_MATCHING
   PATTERN "**/*.h")
 
```

Comment:
Could we also move the following to svf-llvm/CMakeList.txt?

# Compile extapi.c to extapi.bc
find_path(LLVM_CLANG_DIR
        NAMES clang llvm
        HINTS ${LLVM_DIR} ENV LLVM_DIR
        PATH_SUFFIXES bin)
add_custom_target(extapi_ir ALL
    COMMAND ${LLVM_CLANG_DIR}/clang -w -S -c -Xclang -disable-O0-optnone -fno-discard-value-names -emit-llvm ${PROJECT_SOURCE_DIR}/svf-llvm/lib/extapi.c -o ${PROJECT_BINARY_DIR}/svf-llvm/extapi.bc
    DEPENDS ${PROJECT_SOURCE_DIR}/svf-llvm/lib/extapi.c
)

Reply:
Done.

Reply:
I also moved the installation commands to the subdirectory so the `SvfLLVM` target isn't referenced at all in the top-level `CMakeLists.txt` anymore.

While doing that I also fixed the remaining installation predicates:
1. `<svf_root>/svf/include` (i.e. its contents) is installed to `<install_prefix>/include/svf/`
2. `<svf_root>/svf-llvm/include` (i.e. the `SVF-LLVM` directory therein) is installed to `<install_prefix>/include/svf/`
3. `<svf_root/[Debug|Release]-Build/include/Util/config.h` is installed to `<install_prefix>/include/svf/Util/config.h`
4. `<svf_root/[Debug|Release]-Build/svf/libSvfCore.[a|so]` is installed to `<install_prefix>/lib/libSvfCore.[a|so]`
5. `<svf_root/[Debug|Release]-Build/svf-llvm/libSvfLLVM.[a|so]` is installed to `<install_prefix>/lib/libSvfLLVM[a|so]`
6. `<svf_root>/[Debug|Release]-Build/svf-llvm/extapi.bc` is installed to `<install_prefix>/include/svf/SVF-LLVM/extapi.bc` (I thought this was the best place as placing it by itself in `<install_prefix>/lib/extapi.bc` so that it's placed alongside `libSVF.[a|so]` seemed inconvenient)
7. Tools from `<svf_root>/svf-llvm/tools/...` (built to `<svf_root>/[Debug|Release]-Build/bin/...`) are installed to `<install_prefix>/bin/...`

This should be unnecessary for the way SVF is currently designed to be used (i.e. through the `build.sh` and `setup.sh` scripts), but it allows users who want to install SVF to a specific location to not have to manually link/include the different directories/library files so explicitly.